common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -761,6 +764,13 @@ def _render_student_view_for_items(self, context, display_items, fragment, view=
                 contains_content_type_gated_content = content_type_gating_service.check_children_for_content_type_gating_paywall(  # pylint:disable=line-too-long
                     item, self.course_id
                 ) is not None
+
+            has_integrity_signature = False
+            if is_integrity_signature_enabled():
+                integrity_signature = get_integrity_signature(context['username'], str(self.course_id))
```

Comment:
We also want to check if the module contains graded content, see https://github.com/edx/edx-platform/blob/e66e43c5d2d452ec3a2c609fe26dbe7b4abba565/openedx/features/content_type_gating/services.py#L50 which calls https://github.com/edx/edx-platform/blob/e66e43c5d2d452ec3a2c609fe26dbe7b4abba565/openedx/features/content_type_gating/services.py#L36 as an example

Reply:
Instead of using the toggle, `has_access` should be used here (as that covers enrollment, staff, and now honor code access limitations)

Reply:
That makes sense, thank you!